server/src/main/java/org/candlepin/async/JobMessageDispatcher.java

```diff
@@ -14,13 +14,159 @@
  */
 package org.candlepin.async;
 
+import org.candlepin.common.config.Configuration;
+import org.candlepin.messaging.CPMException;
+import org.candlepin.messaging.CPMMessage;
+import org.candlepin.messaging.CPMProducer;
+import org.candlepin.messaging.CPMProducerConfig;
+import org.candlepin.messaging.CPMSession;
+import org.candlepin.messaging.CPMSessionConfig;
+import org.candlepin.messaging.CPMSessionFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
 
 
 /**
- * The JobMessageDispatcher interface defines a method for sending a job message to another
- * queue or underlying messaging system.
+ * The JobMessageDispatcher is responsible for managing sessions to the backing messaging system,
+ * and serializing messages as they're sent.
  */
-public interface JobMessageDispatcher {
+public class JobMessageDispatcher {
+    private static Logger log = LoggerFactory.getLogger(JobMessageDispatcher.class);
+
+    private static final String JOB_KEY_MESSAGE_PROPERTY = "job_key";
+    private static final String JOB_MESSAGE_ADDRESS = "job";
+
+    private final Configuration config;
+    private final CPMSessionFactory cpmSessionFactory;
+    private final ObjectMapper objMapper;
+
+    private boolean initialized;
+
+    private CPMSession cpmSession;
+    private CPMProducer cpmProducer;
+
+    /**
+     * Creates a new JobMessageDispatcher instance for sending job messages to the backing
+     * message bus.
+     *
+     * @param cpmSessionFactory
+     *  A CPMSessionFactory instance for creating messaging sessions
+     *
+     * @param objMapper
+     *  An ObjectMapper instance for serializing job messages prior to sending them to the
+     *  message bus
+     */
+    @Inject
+    public JobMessageDispatcher(Configuration config, CPMSessionFactory cpmSessionFactory,
+        ObjectMapper objMapper) {
+
+        this.config = Objects.requireNonNull(config);
+        this.cpmSessionFactory = Objects.requireNonNull(cpmSessionFactory);
+        this.objMapper = Objects.requireNonNull(objMapper);
+
+        this.initialized = false;
+    }
+
+    /**
+     * Initializes this message dispatcher, specfiying the job manager that will be dispatching
+     * messages
+     *
+     * @param manager
+     *  the JobManager that will be dispatching messages
+     *
+     * @throws IllegalStateException
+     *  if this message receiver has already been initialized
+     *
+     * @throws IllegalArgumentException
+     *  if the provided job manager is null
+     */
+    public synchronized void initialize(JobManager manager) {
```

Comment:
This seems to not do anything. At first glance at least, it seems like this method, the `initialized` var, and `isInitialized` are no longer necessary?

Reply:
Maybe not. These were added before any actual code was written, so they largely live on due to the interface not changing after the fact and for consistency; though I suppose they could be dropped.